src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -7975,36 +7946,10 @@ void SketchObject::rebuildExternalGeometry()
                                     }
                                 }
                                 else if (projCurve.GetType() == GeomAbs_BSplineCurve) {
-                                    // Unfortunately, a normal projection of a circle can also give
-                                    // a Bspline Split the spline into arcs
-                                    GeomConvert_BSplineCurveKnotSplitting bSplineSplitter(
-                                        projCurve.BSpline(), 2);
-                                    // int s = bSplineSplitter.NbSplits();
-                                    if ((curve.GetType() == GeomAbs_Circle)
-                                        && (bSplineSplitter.NbSplits() == 2)) {
-                                        // Result of projection is actually a circle...
-                                        TColStd_Array1OfInteger splits(1, 2);
-                                        bSplineSplitter.Splitting(splits);
-                                        gp_Pnt p1 = projCurve.Value(splits(1));
-                                        gp_Pnt p2 = projCurve.Value(splits(2));
-                                        gp_Pnt p3 = projCurve.Value(0.5 * (splits(1) + splits(2)));
-                                        GC_MakeCircle circleMaker(p1, p2, p3);
-                                        Handle(Geom_Circle) circ = circleMaker.Value();
-                                        Part::GeomCircle* circle = new Part::GeomCircle();
-                                        circle->setRadius(circ->Radius());
-                                        gp_Pnt center = circ->Axis().Location();
-                                        circle->setCenter(
-                                            Base::Vector3d(center.X(), center.Y(), center.Z()));
-
-                                        GeometryFacade::setConstruction(circle, true);
-                                        ExternalGeo.push_back(circle);
-                                    }

```

Comment:
This seems like useful functionality, was this moved somewhere else?

Reply:
This comment `Result of projection is actually a circle...` may be wrong. A circle's projection on to a plane can be an ellipse or even a line.

Reply:
Circles and ellipses are what could project into a circle and these cases are fully handled above. This means this code is dead code or worse it's running for cases it wasn't intended to handle (the comment talks about handling "a normal projection of a circle"). Note parabolas or hyperbolas can't project into a full circle.

Reply:
Apologies for my ignorance, but I assumed `curve` is the original shape, and `projCurve` is the shape as projected on to the plane. If it is vice versa, your explanation is perfectly valid[1].

If my assumption is correct, this check is likely to attempt to fix when the projections are converted into B-splines. OCCT may be doing this under certain conditions which we don't know of right now, but which the original author of this chunk was aware of. If you try to create an external curve from a B-spline, this curve has a higher dimension than the original curve, so it is likely that such changes may be done in the background for (arcs of) circles as well.

[1] Note, however, that under "perspective" projections (when lines are drawn to a point instead of perpendicular to plane) _arcs of_ parabolas and hyperbolas can result in _arcs of_ circle.

Reply:
`Curve` is indeed the orginal shape and `projCurve` is the shape as projected.

To understand what I'm saying check the surrounding code. We have:
```C++
if (curve.GetType() == GeomAbs_Circle) {
    [...]
}
else if (curve.GetType() == GeomAbs_Ellipse) {
    [...]
}
else {
    [...]
                if ((curve.GetType() == GeomAbs_Circle)
                                        && (bSplineSplitter.NbSplits() == 2)) {
                      [.. dead code ..]
                }
}
```

The `curve.GetType() == GeomAbs_Circle` case has already been handled so I'm just removing dead code.

Reply:
> The `curve.GetType() == GeomAbs_Circle` case has already been handled so I'm just removing dead code.

In that case consider moving the case where `projCurve.GetType() == GeomAbs_BSpline` into that statement.

Reply:
To where and to handle what case? There is no `projCurve` in this section:
```
if (curve.GetType() == GeomAbs_Circle) {
    [...]
}
```